components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/state/OAuthRequestStateValidator.java

```diff
@@ -132,12 +133,15 @@ private void validateInputParameters(OAuthMessage oAuthMessage) throws InvalidRe
                     OAuth2ErrorCodes.INVALID_REQUEST, OAuth2ErrorCodes.OAuth2SubErrorCodes.INVALID_CLIENT);
         }
 
-        if (StringUtils.isBlank(oAuthMessage.getRequest().getParameter(REDIRECT_URI))) {
-            if (log.isDebugEnabled()) {
-                log.debug("Redirect URI is not present in the authorization request");
+        if(
+        !StringUtils.equals(oAuthMessage.getRequest().getParameter(Constants.RESPONSE_TYPE), Constants.DEVICE)) {
+            if (StringUtils.isBlank(oAuthMessage.getRequest().getParameter(REDIRECT_URI))) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Redirect URI is not present in the authorization request");
```

Comment:
Can we improve this debug log?

Reply:
It was already in identity-inbound-auth-oauth ayya. I just add if (line 136) command